doc/sphinx/developer_manual/contributing_to_the_code/tips_and_tricks.md

```diff
@@ -4,4 +4,79 @@ Tips and tricks
 
 ```{todo}
 Give some tick and tricks, like start copying plugins which already so almost what you want.
-```
\ No newline at end of file
+```
+
+# Tricks to Run Tests
+
+## Trick 1: Run All Tests
+
+This command would compile and run all the tests.
+
+    make test
+
+Not so much information is given for the failing tests, so the next step is needed for more information.
+
+## Trick 2: Run one test with more information
+
+This command would run one test and print more information to the screen:
+
+    ctest -V -R name_of_test
+
+This is useful to check for failing tests. At the end of the message, the differences between the generated file of the test and the standard file would be shown.
+
+```{todo}
+Give a workflow to generate new tests
+```
+
+## Trick 3: Run the unit_tests separately
+
+In trick 1, the unit_tests are grouped as one big test. Again, not so much information is given about the individual unit tests.
+Once the tests are compiled in step 1, a separate executable could be found in the build folder which provides more information regarding the unit tests:
+
+    ./build/unit_tests
+
+```{todo}
+Give a workflow to generate new unit tests
+```
+
+## Trick 4: Interact with visualization when generating new tests
+
+The tests in the WorldBuilder contain a wb file for the setup and a dat file for the mesh point. 
+For example, the app_spherical_2d test contains two files:
+
+    tests/gwb-dat/app_spherical_2d.wb
+    tests/gwb-dat/app_spherical_2d.dat
+
+And a folder:
+
+     tests/gwb-dat/app_spherical_2d
+
+
+The first few lines of the contents in the dat file are:
+
+    # This is a comment in the data
```

Comment:
I am surprised it already automatically makes code blocks from this, but it might be nice to make it a bit more formal and add line numbering, like this (I haven't tested it, but I think it should work):

```
```{code-block} python
---
lineno-start: 1
---
{
    "version":"0.3",
    "coordinate system":{"model":"cartesian"},
    "features":[ ]
}
\```
```

(without the \, but otherwise github gets confussed)

Reply:
added